CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -788,28 +813,13 @@ + (void)subscribe:(CPHandleSubscribedBlock)subscribedBlock {
                             
                             [self getChannelConfig:^(NSDictionary* channelConfig) {
                                 if (channelConfig != nil && ([channelConfig valueForKey:@"confirmAlertHideChannelTopics"] == nil || ![[channelConfig valueForKey:@"confirmAlertHideChannelTopics"] boolValue])) {
-                                    NSArray* channelTopics = [channelConfig valueForKey:@"channelTopics"];
-                                    if (channelTopics != nil && [channelTopics count] > 0) {
-                                        NSArray* topics = [self getSubscriptionTopics];
-                                        
-                                        if (!topics || [topics count] == 0) {
-                                            NSMutableArray* selectedTopicIds = [[NSMutableArray alloc] init];
-                                            for (id channelTopic in channelTopics) {
-                                                if (channelTopic != nil && ([channelTopic valueForKey:@"defaultUnchecked"] == nil || ![[channelTopic valueForKey:@"defaultUnchecked"] boolValue])) {
-                                                    [selectedTopicIds addObject:[channelTopic valueForKey:@"_id"]];
-                                                }
-                                            }
-                                            if ([selectedTopicIds count] > 0) {
-                                                [self setSubscriptionTopics:selectedTopicIds];
-                                            }
-                                        }
-                                        
-                                        NSUserDefaults* userDefaults = [NSUserDefaults standardUserDefaults];
-                                        [userDefaults setBool:YES forKey:@"CleverPush_TOPICS_DIALOG_PENDING"];
-                                        [userDefaults synchronize];
-                                        
-                                        [self showPendingTopicsDialog];
+                                    if (![self isSubscribed]) {
```

Comment:
Why do we need this check?

Reply:
@gbhrdt because inside that `if` condition I have executed function `[self initialisedTopicDialogData:channelConfig];` and in that function `(initialisedTopicDialogData:channelConfig)` we have checked that the it will only refill the topics value when count is zero
`NSArray* topics = [self getSubscriptionTopics];`
        
`if (!topics || [topics count] == 0) {` but topics's count will be zero once you deselect all the topics so it will be populate the default selected topics in the topic dialog instead of the unchecked all, So that is why i put the check of the `(![self isSubscribed])` at that part so if it is not subscribed yet then & then it will be populated the default check in the Topic dialog.